telemetry/Tracks/class-tracks.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Telemetry: Tracks class
+ *
+ * @package Automattic\VIP\Telemetry
+ */
+
+declare(strict_types=1);
+
+namespace Automattic\VIP\Telemetry;
+
+use WP_Error;
+
+/**
+ * This class comprises the mechanics of sending events to the Automattic Tracks
+ * system.
+ */
+class Tracks extends Telemetry_System {
+	/**
+	 * Registers the events into WordPress hooks to activate tracking.
+	 */
+	public function run(): void {
+		$this->activate_tracking();
+	}
+
+	/**
+	 * Records an event to Tracks by using the Tracks pixel.
+	 *
+	 * Depending on the current context, the pixel will be recorded
+	 * synchronously (as a GET request) or as asynchronously (as an injected
+	 * pixel into the page's footer).
+	 *
+	 * If the event doesn't pass validation, it gets silently discarded.
+	 *
+	 * @param string                            $event_name The event name. Must be snake_case.
+	 * @param array<string, mixed>|array<empty> $event_properties Any additional properties to include with the event.
+	 *                                                            Key names must be lowercase and snake_case.
+	 * @return bool|WP_Error True if recording the event succeeded.
+	 *                       False if telemetry is disabled.
+	 *                       WP_Error if recording the event failed.
+	 */
+	public function record_event(
+		string $event_name,
+		array $event_properties = array()
+	) {
+		$event = new Tracks_Event( $event_name, $event_properties );
```

Comment:
If the Tracks Event had some errors, we should log those somewhere so we have visibility into failures.

Reply:
I think my changes in 1b308c7646759ea9ad36c94fac528bfd669cf294 should cover this